contributing/adr/0005-use-erb-for-component-templates.md

```diff
@@ -0,0 +1,32 @@
+# 5. Use ERB for component templates
+
+Date: 2022-07-26
+
+## Status
+
+Proposed
+
+## Context
+
+We've seen output buffer issues when using out components outside of Rails within our Bridgetown docs. This happens where the components using a Haml template and where there is any kind of nested content block (e.g. tag helpers or composition). See https://github.com/citizensadvice/design-system/pull/2207 for an example of the issue and https://github.com/citizensadvice/design-system/pull/2207 demonstrating that using ERB for the template resolves any issues.
+
+It's possible to work around some of the issues by adjusting the Haml templates with delicate use of `capture_haml` or by adjusting other parts of the templates on a case by case basis but this feels fraught.
+
+## Decision
+
+We will use ERB for our component templates. Initially we will migrate affected templates over to ERB with the goal of having migrated all Haml templates over to ERB and to have removed `haml-rails` as a dependency by version 6 of the design system.
+
+This decision only effects the internal template language used for components. It does not prevent consumers of the design system from using Haml for any templates.
+
+## Consequences
+
+**Pros:**
+
+- Makes it possible to use all components in our docs without trade-offs and simplify our Bridgetown integration
+- ERB is a core part of ruby so has a higher likelihood of working without caveats and be officially supported
+- Allows us to eventually remove `haml-rails` from our dependencies
+
+**Cons:**
+
+- Limits us to only using ERB internally
+- When migrating components over from consumers into the design system
```

Comment:
@davidsauntson Does this seem representative for the cons. The first one is the immediate trade-off but felt worth capturing the fact that there'd be an extra step when porting components from public-website.